scripts/k6/tenant.js

```diff
@@ -0,0 +1,809 @@
+      // console.log(`event data=${event.data}`);
+      const eventData = JSON.parse(event.data);
+      if (eventData.topic === "credentials" && eventData.payload.state === "offer-received") {
+        check(eventData, {
+          "Event received": (e) => e.payload.state === "offer-received",
+        });
+        eventReceived = true;
+        client.close();
+      }
+    });
+
+    client.on("error", function (e) {
+      console.log("An unexpected error occurred: ", e.error());
+      client.close();
+    });
+  });
+
+  check(response, { "SSE connection established": (r) => r && r.status === 200 });
+
+  // Wait for the event to be received or a maximum duration
+  const maxDuration = 10; // 10 seconds
+  const checkInterval = 1; // 1 second
+  let elapsedTime = 0;
+
+  while (!eventReceived && elapsedTime < maxDuration) {
+    console.log(`Waiting for event... Elapsed time: ${elapsedTime}ms`);
+    elapsedTime += checkInterval;
+    sleep(checkInterval);
+  }
+
+  return eventReceived;
+}
+
+export function waitForSSEEventConnection(holderAccessToken, holderWalletId, invitationConnectionId) {
+  const sseUrl = `${__ENV.CLOUDAPI_URL}/tenant/v1/sse/${holderWalletId}/connections/connection_id/${invitationConnectionId}/completed`;
+  const headers = {
+    "x-api-key": holderAccessToken,
+  };
+
+  let eventReceived = false;
+
+  const response = sse.open(sseUrl, {
+    headers,
+    tags: { "k6_sse_tag": "connection_ready" },
+  }, function (client) {
+    client.on("event", function (event) {
+      // console.log(`event data=${event.data}`);
+      const eventData = JSON.parse(event.data);
+      if (eventData.topic === "connections" && eventData.payload.state === "completed") {
+        check(eventData, {
+          "Event received": (e) => e.payload.state === "completed",
+        });
+        eventReceived = true;
+        client.close();
+      }
+    });
+
+    client.on("error", function (e) {
+      console.log("An unexpected error occurred: ", e.error());
+      client.close();
+    });
+  });
+
+  check(response, { "SSE connection established": (r) => r && r.status === 200 });
+
+  // Create random number between 1 and 3
+  // const random = Math.floor(Math.random() * 3) + 1;
+
+  // Wait for the event to be received or a maximum duration
+  const maxDuration = 10; // 10 seconds
+  const checkInterval = 1; // 1 second
+  let elapsedTime = 0;
+
+  while (!eventReceived && elapsedTime < maxDuration) {
+    console.log(`Waiting for event... Elapsed time: ${elapsedTime}ms`);
+    elapsedTime += checkInterval;
+    sleep(checkInterval);
+  }
+  // if number equals 1, sleep for 1 second
+  // if (random === 1) {
+  //   console.log("Sleeping for 1 second");
+  //   eventReceived = false
+  // }
+  return eventReceived;
+}
+
+export function getCredentialDefinitionId(bearerToken, issuerAccessToken, credDefTag) {
+  const url = `${__ENV.CLOUDAPI_URL}/tenant/v1/definitions/credentials?schema_version=0.1.0`;
```

Comment:
minor note: this schema_version=0.1.0 seems it should use the SCHEMA_VERSION env var, which equals 0.1.0

Reply:
It should be passed to the function. Hardcoded the schema stuff before when it was still being created outside of k6. Will fix that as I add flows for schema creation.